File: codegen/results/v001/qwen3.5_4b_nothink/20260408_194733/go/005_unit_test_writer/solution.go
package main

import (
	"fmt"
	"math"
)

// calculateDiscount calculates a bulk discount
func calculateDiscount(price float64, quantity int) float64 {
	if quantity < 10 {
		return price * float64(quantity)
	} else if quantity < 50 {
		return price * float64(quantity) * 0.9
	}
	return price * float64(quantity) * 0.75
}

func main() {
	testCases := []struct {
		desc     string
		price    float64
		quantity int
		expected float64
	}{
		{desc: "single item price", price: 10.0, quantity: 1, expected: 10.0},
		{desc: "boundary below 10", price: 10.0, quantity: 9, expected: 90.0},
		{desc: "boundary at 10", price: 10.0, quantity: 10, expected: 90.0},
		{desc: "boundary below 50", price: 10.0, quantity: 49, expected: 441.0},
		{desc: "boundary at 50", price: 10.0, quantity: 50, expected: 375.0},
		{desc: "large quantity", price: 10.0, quantity: 100, expected: 750.0},
	}

	for _, tc := range testCases {
		result := calculateDiscount(tc.price, tc.quantity)
		// Use a small epsilon for float comparison
		epsilon := 1e-6
		isMatch := math.Abs(result-tc.expected) < epsilon

		if isMatch {
			fmt.Printf("PASS: %s\n", tc.desc)
		} else {
			fmt.Printf("FAIL: %s (expected: %.1f, got: %.1f)\n", tc.desc, tc.expected, result)
		}
	}
}

File: codegen/results/v001/qwen3.5_4b_nothink/20260408_194733/typescript/005_unit_test_writer/solution.ts
function calculateDiscount(price: number, quantity: number): number {
    if (quantity < 10) return price * quantity;
    else if (quantity < 50) return price * quantity * 0.9;
    else return price * quantity * 0.75;
}

// Run tests
const tests = [
    {
        desc: "Quantity less than 10 (boundary)",
        price: 10,
        quantity: 9,
        expected: 90,
    },
    {
        desc: "Quantity at 10 (boundary)",
        price: 10,
        quantity: 10,
        expected: 90,
    },
    {
        desc: "Quantity at 49 (boundary)",
        price: 10,
        quantity: 49,
        expected: 441, // 49 * 10 * 0.9
    },
    {
        desc: "Quantity at 50 (boundary)",
        price: 10,
        quantity: 50,
        expected: 375, // 50 * 10 * 0.75
    },
];

let failed = false;

for (const test of tests) {
    const result = calculateDiscount(test.price, test.quantity);
    if (result === test.expected) {
        console.log(`PASS: ${test.desc}`);
    } else {
        console.log(`FAIL: ${test.desc} (expected: ${test.expected}, got: ${result})`);
        failed = true;
    }
}

if (failed) {
    process.exit(1);
}